src/main/java/frc/robot/Robot.java

```diff
@@ -291,37 +293,43 @@ private void spinControlPanel() {
         }
       }
 
-      String colorString;
+      String detectedColorString;
       Color detectedColor = colorSensor.getColor();
       ColorMatchResult match = colorMatcher.matchClosestColor(detectedColor);
       if (match.color == kBlueTarget) {
-        colorString = "Blue";
+        detectedColorString = "Blue";
       } else if (match.color == kRedTarget) {
-        colorString = "Red";
+        detectedColorString = "Red";
       } else if (match.color == kGreenTarget) {
-        colorString = "Green";
+        detectedColorString = "Green";
       } else if (match.color == kYellowTarget) {
-        colorString = "Yellow";
+        detectedColorString = "Yellow";
       } else {
-        colorString = "Unknown";
+        detectedColorString = "Unknown";
       }
-      detectedColorEntry.setString(colorString);
+      String lastDetectedColorString = detectedColorEntry.toString(); 
+      detectedColorEntry.setString(detectedColorString);
       confidenceEntry.setDouble(match.confidence);
-
+      
       int doubleContolPanelSpinAmount = controlPanelSpinAmount * 2;
-      if (targetControlPanelColor == "Green" && (colorString != "Green" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      } else if (targetControlPanelColor == "Red" && (colorString != "Red" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      } else if (targetControlPanelColor == "Blue" && (colorString != "Blue" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      } else if (targetControlPanelColor == "Yellow" && (colorString != "Yellow" || doubleContolPanelSpinAmount > 0)) {
-        // @TODO: Add wheel spinner code (currently waiting for more details on this).
-      }
+      turnControlPanel(detectedColorString, targetControlPanelColor, lastDetectedColorString,
```

Comment:
IMO, rather than passing in `lastDetectedColorString` here, it would be cleaner to return from this method a couple of lines above from this, if `detectedColorString == lastDetectedColorString`. If we have been on the same color, then it's not necessary to go through the logic in `turnControlPanel` again.